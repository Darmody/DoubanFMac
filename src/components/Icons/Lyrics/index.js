/* eslint-disable */
import React, { PureComponent } from 'react'
import styled from 'styled-components'

const Icon = styled.svg`
  ${props => props.styles}
`

export default class IconLyrics extends PureComponent {
  render() {
    return (
      <Icon
        styles={this.props.styles}
        title="Lyrics" viewBox="0 0 20 20" height="20" width="20"
      >
        <desc>Icon</desc><g id="icon" fill="none" fillRule="evenodd"><g id="ci" fill="#B9B9B9"><path d="M2 7.5c0-.276.229-.5.5-.5h11c.276 0 .5.232.5.5 0 .276-.229.5-.5.5h-11a.505.505 0 0 1-.5-.5zm1 3c0-.276.215-.5.498-.5h4.004c.275 0 .498.232.498.5 0 .276-.215.5-.498.5H3.498A.504.504 0 0 1 3 10.5zm0 3c0-.276.215-.5.498-.5h4.004c.275 0 .498.232.498.5 0 .276-.215.5-.498.5H3.498A.504.504 0 0 1 3 13.5zm-1 3c0-.276.233-.5.503-.5h5.994c.278 0 .503.232.503.5 0 .276-.233.5-.503.5H2.503A.507.507 0 0 1 2 16.5zm1-12c0-.276.215-.5.498-.5h4.004c.275 0 .498.232.498.5 0 .276-.215.5-.498.5H3.498A.504.504 0 0 1 3 4.5zM9.5 4c-.303 0-.5.255-.5.5s.197.5.5.5H16v11l-3.5.044c-.328 0-.5.15-.5.456 0 .308.172.5.5.5H16c.497 0 1-.5 1-1V5c0-.5-.505-1-1-1H9.5zm4.5 6.604a.601.601 0 0 0-.596-.604H9.597a.602.602 0 0 0-.597.604v2.791c0 .333.269.605.597.605h3.807a.602.602 0 0 0 .596-.605v-2.79zM10 11h3v2h-3v-2z" id="Combined-Shape"></path></g></g>
      </Icon>
    )
  }
}
