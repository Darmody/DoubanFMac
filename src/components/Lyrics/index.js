import React, { PureComponent } from 'react'
import styled from 'styled-components'

const ipc = require('electron').ipcRenderer

const Lyrics = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

const LyricsList = styled.div`
  text-align: center;
  padding: 3rem 0;
  height: 25.813rem;
  overflow: scroll;
`

const Action = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: white;
  height: 9.375rem;
`

const CloseLyrics = styled.a`
  color: #8f8e94;
  font-size: .813rem;
  font-weight: 400;
  line-height: 1.75rem;
  text-decoration: none;
  cursor: pointer;
`

const Line = styled.p`
  color: #8f8e94;
  font-size: .813rem;
  font-weight: 400;
  line-height: 1.75rem;
  min-height: 1rem;
  user-select: text;
  cursor: text;
`

export default class LyricsContainer extends PureComponent {
  handleClose = () => {
    ipc.send('lyricsWindow', 'close')
  }

  render() {
    return (
      <Lyrics>
        <LyricsList>
          <Line>Pick apart</Line>
          <Line>The pieces of your heart</Line>
          <Line>And let me peer inside</Line>
          <Line>Let me in</Line>
          <Line>Where only your thoughts have been</Line>
          <Line>Let me occupy your mind</Line>
          <Line>As you do mine</Line>
          <Line>Your heart’s a mess</Line>
          <Line>You won’t admit to it</Line>
          <Line>It makes no sense</Line>
          <Line>But I’m desperate to connect</Line>
          <Line>And you, you can’t live like this</Line>
          <Line>You have lost</Line>
          <Line>Too much love</Line>
          <Line>To fear, doubt and distrust</Line>
          <Line>(It’s not enough)</Line>
          <Line>You just threw away the key</Line>
          <Line>To your heart</Line>
          <Line>You don’t get burned</Line>
          <Line>(’Cause nothing gets through)</Line>
          <Line>It makes it easier</Line>
          <Line>(Easier on you)</Line>
          <Line>But that much more difficult for me</Line>
          <Line>To make you see…</Line>
          <Line>Love ain’t fair</Line>
          <Line>So there you are</Line>
          <Line>My love</Line>
          <Line>Your heart’s a mess</Line>
          <Line>You won’t admit to it</Line>
          <Line>It makes no sense</Line>
          <Line>But I’m desperate to connect</Line>
          <Line>And you, you can’t live like this</Line>
          <Line>Love ain’t safe</Line>
          <Line>You won’t get hurt if you stay chaste</Line>
          <Line>So you can wait</Line>
          <Line>But I don’t wanna waste my love</Line>
        </LyricsList>
        <Action>
          <CloseLyrics
            href="javascript:void(0);"
            onClick={this.handleClose}
          >
            关闭歌词
          </CloseLyrics>
        </Action>
      </Lyrics>
    )
  }
}
