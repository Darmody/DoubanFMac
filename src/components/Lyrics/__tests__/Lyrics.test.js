import React from 'react'
import renderer from 'react-test-renderer'
import Lyrics from 'components/Lyrics'

describe('Component::Lyrics', () => {
  test('渲染正确的 DOM 结构', () => {
    const component = renderer.create(<Lyrics />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
