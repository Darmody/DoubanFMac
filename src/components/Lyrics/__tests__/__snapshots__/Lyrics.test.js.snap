// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Lyrics 渲染正确的 DOM 结构 1`] = `
<div
  className="dJEGex"
>
  <div
    className="fHcqxU"
  >
    <p
      className="eAQULc"
    >
      Pick apart
    </p>
    <p
      className="eAQULc"
    >
      The pieces of your heart
    </p>
    <p
      className="eAQULc"
    >
      And let me peer inside
    </p>
    <p
      className="eAQULc"
    >
      Let me in
    </p>
    <p
      className="eAQULc"
    >
      Where only your thoughts have been
    </p>
    <p
      className="eAQULc"
    >
      Let me occupy your mind
    </p>
    <p
      className="eAQULc"
    >
      As you do mine
    </p>
    <p
      className="eAQULc"
    >
      Your heart’s a mess
    </p>
    <p
      className="eAQULc"
    >
      You won’t admit to it
    </p>
    <p
      className="eAQULc"
    >
      It makes no sense
    </p>
    <p
      className="eAQULc"
    >
      But I’m desperate to connect
    </p>
    <p
      className="eAQULc"
    >
      And you, you can’t live like this
    </p>
    <p
      className="eAQULc"
    >
      You have lost
    </p>
    <p
      className="eAQULc"
    >
      Too much love
    </p>
    <p
      className="eAQULc"
    >
      To fear, doubt and distrust
    </p>
    <p
      className="eAQULc"
    >
      (It’s not enough)
    </p>
    <p
      className="eAQULc"
    >
      You just threw away the key
    </p>
    <p
      className="eAQULc"
    >
      To your heart
    </p>
    <p
      className="eAQULc"
    >
      You don’t get burned
    </p>
    <p
      className="eAQULc"
    >
      (’Cause nothing gets through)
    </p>
    <p
      className="eAQULc"
    >
      It makes it easier
    </p>
    <p
      className="eAQULc"
    >
      (Easier on you)
    </p>
    <p
      className="eAQULc"
    >
      But that much more difficult for me
    </p>
    <p
      className="eAQULc"
    >
      To make you see…
    </p>
    <p
      className="eAQULc"
    >
      Love ain’t fair
    </p>
    <p
      className="eAQULc"
    >
      So there you are
    </p>
    <p
      className="eAQULc"
    >
      My love
    </p>
    <p
      className="eAQULc"
    >
      Your heart’s a mess
    </p>
    <p
      className="eAQULc"
    >
      You won’t admit to it
    </p>
    <p
      className="eAQULc"
    >
      It makes no sense
    </p>
    <p
      className="eAQULc"
    >
      But I’m desperate to connect
    </p>
    <p
      className="eAQULc"
    >
      And you, you can’t live like this
    </p>
    <p
      className="eAQULc"
    >
      Love ain’t safe
    </p>
    <p
      className="eAQULc"
    >
      You won’t get hurt if you stay chaste
    </p>
    <p
      className="eAQULc"
    >
      So you can wait
    </p>
    <p
      className="eAQULc"
    >
      But I don’t wanna waste my love
    </p>
  </div>
  <div
    className="iWMXlF"
  >
    <a
      className="iYKYhB"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      关闭歌词
    </a>
  </div>
</div>
`;
